core/src/com/unciv/logic/map/mapunit/MapUnit.kt

```diff
@@ -343,8 +344,23 @@ class MapUnit : IsPartOfGameInfoSerialization {
         if (updateCivViewableTiles && oldViewableTiles != viewableTiles
                 // Don't bother updating if all previous and current viewable tiles are within our borders
                 && (oldViewableTiles.any { it !in civ.cache.ourTilesAndNeighboringTiles }
-                        || viewableTiles.any { it !in civ.cache.ourTilesAndNeighboringTiles }))
+                        || viewableTiles.any { it !in civ.cache.ourTilesAndNeighboringTiles })) {
+
+            val newlyExploredTiles = viewableTiles.filter {
+                !it.isExplored(civ)
+            }
+            for (tile in newlyExploredTiles) {
+                // Include tile in the state for correct RNG seeding
+                val state = StateForConditionals(civInfo=civ, unit=this, tile=tile);
+                for (unique in getTriggeredUniques(UniqueType.TriggerUponDiscoveringTile, state)) {
+                    if (unique.conditionals.any { it.type == UniqueType.TriggerUponDiscoveringTile
+                        && tile.matchesFilter(it.params[0], civ) })
+                        UniqueTriggerActivation.triggerUnitwideUnique(unique, this)
+                }
+            }
```

Comment:
```suggestion
            if (newlyExploredTiles.isNotEmpty){
                val unfilteredTriggeredUniques = getTriggeredUniques(UniqueType.TriggerUponDiscoveringTile, StateForConditionals.IgnoreConditionals)
                for (tile in newlyExploredTiles) {
                    // Include tile in the state for correct RNG seeding
                    val state = StateForConditionals(civInfo=civ, unit=this, tile=tile);
                    for (unique in unfilteredConditionals) {
                        if (unique.conditionals.any { it.type == UniqueType.TriggerUponDiscoveringTile
                            && tile.matchesFilter(it.params[0], civ) }
                                && unique.conditionalsApply(state))
                            UniqueTriggerActivation.triggerUnitwideUnique(unique, this)
                    }
                }
            }
```

I think this is what SomeTrog was aiming for, tbh my first thought was also to get triggerables once

Reply:
You're right, it needs a .toList() :)

Reply:
That's fair, given that this trigger is pretty niche, the game would calculate newlyExploredTiles far more often than needed. I've implemented a fix with both your suggestions, thanks again for your comments!